lib/mongo/crypt/auto_encrypter.rb

```diff
@@ -66,14 +72,52 @@ def setup_encrypter(options = {})
                                 monitoring_io: mongocryptd_client_monitoring_io,
                               )
 
+        unless @encryption_options[:mongocryptd_bypass_spawn] || !mongocryptd_client_monitoring_io
+          self.spawn_mongocryptd
+        end
+
         # TODO: use all the other options for auto-encryption/auto-decryption
       end
 
+      # Spawn a new mongocryptd process using the mongocryptd_spawn_path
+      # and mongocryptd_spawn_args passed in through the extra auto
+      # encrypt options. Stdout and Stderr of this new process are written
+      # to /dev/null.
+      #
+      # @return [ Integer ] The process id of the spawned process
+      def spawn_mongocryptd
+        @mongocryptd_pid = Process.spawn(
+                            @encryption_options[:mongocryptd_spawn_path],
+                            *@encryption_options[:mongocryptd_spawn_args],
+                            [:out, :err]=>'/dev/null'
+                          )
+      end
+
+      # Abort the spawned mongocryptd process. Will not raise an error,
+      # even if the process has already been terminated.
+      #
+      # @return [ true ] Always true
+      def kill_mongocryptd
+        return true if @mongocryptd_pid.nil?
+
+        begin
+          Process.kill('ABRT', @mongocryptd_pid)
```

Comment:
I don't know much about process kill signals. Abort seemed like the appropriate signal to use in this case, but I'm open to guidance.